### ISSUE TITLE
LocallyConnected2D should raise error with negative strides

### ISSUE THREAD

**Issue by lugalUrim** [keras-team/keras#15368]
**System information**.
- Have I written custom code (as opposed to using a stock example script provided in Keras): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 18.04
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): nightly
- Python version: 3.7.11
- Bazel version (if compiling from source): n/a
- GPU model and memory: n/a
- Exact command to reproduce: n/a

**Describe the problem**.
`LocallyConnected2D` should raise error when `strides` contains negative values. When `implementation`  is `1` or `3`, it raises `InvalidArgumentError` as expected. However, with `implementation=2`, it runs silently and outputs a `0`-dimentional output.

**Standalone code to reproduce the issue**.
```
import tensorflow as tf
import keras                                                                                                                                                                     
strides = [-4,4,]
layer = keras.layers.LocallyConnected2D(filters=3,kernel_size=3,padding="valid",strides=strides,implementation=2)
input = tf.random.uniform((3, 6, 10, 4), dtype=tf.float32)
res = layer(input)
print(res)
```
Outputs:
```
tf.Tensor([], shape=(3, 0, 2, 3), dtype=float32)
```

Here's a related issue in https://github.com/tensorflow/tensorflow/issues/51624.

**Comment by kumariko** [comment]
@jvishnuvardhan I Was able to reproduce the issue in TF-nightly but not reproduce in TF 2.6 please find the gist [**`here`**](https://colab.research.google.com/gist/kumariko/12051b8435a18b35ea8d056d551aa1a4/51376.ipynb#scrollTo=EVPxM-BVybVW) .Thanks!

**Comment by WingsBrokenAngel** [comment]
Hi @kumariko @lugalUrim , I have resolved your issue by my PR #15356 . Please check it and give me feedback if you have any.

**Comment by jvishnuvardhan** [comment]
@WingsBrokenAngel I think this was resolved with your PR. 

When I run the above code, it throws the following error. Please check the [gist here](https://colab.research.google.com/gist/jvishnuvardhan/c4aafaf8fa95ebae990dbde78253117a/51376.ipynb)

```
-----------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-2-eee29af32b37> in <module>()
      2 import keras
      3 strides = [-4,4,]
----> 4 layer = keras.layers.LocallyConnected2D(filters=3,kernel_size=3,padding="valid",strides=strides,implementation=2)
      5 input = tf.random.uniform((3, 6, 10, 4), dtype=tf.float32)
      6 res = layer(input)

1 frames
/usr/local/lib/python3.7/dist-packages/keras/utils/conv_utils.py in normalize_tuple(value, n, name, allow_zero)
    100     error_msg += (f' including {unqualified_values}'
    101                   f' that does not satisfy the requirement `{req_msg}`.')
--> 102     raise ValueError(error_msg)
    103 
    104   return value_tuple

ValueError: The `strides` argument must be a tuple of 2 integers. Received: [-4, 4] including {-4} that does not satisfy the requirement `>= 0`.
```

I am closing this issue as this was resolved with the merger of PR https://github.com/keras-team/keras/pull/15356
Please feel free to reopen if the issue persists again. Thanks!
